src/taskpane/server.js

```diff
@@ -3228,45 +3928,82 @@ function processSendToSpiraResponse(i, sentToSpira, entriesForExport, artifact,
   if (sentToSpira.error) {
     log.errorCount++;
     response.error = true;
+    var errorCode = 0;
+    if (IS_GOOGLE) {
+      errorCode = response.details.httpCode;
+    }
+    else {
+      errorCode = sentToSpira.errorMessage.status;
+    }
+
+
+
 
     //handling different error messages
-    if (sentToSpira.errorMessage.status == 409) {
+    if (errorCode == 409) {
       response.message = "Concurrency Date conflict: please reload your data and try again.";
     }
     else {
-      if (sentToSpira.errorMessage.status == 400) {
-        try {
-          let parser = new DOMParser();
-          let doc = parser.parseFromString(sentToSpira.errorMessage.response.text, 'text/html');
+      if (errorCode == 400) {
+        if (IS_GOOGLE) {
           try {
-            let errorMessage2 = doc
-              .querySelector('p:nth-of-type(2)')
-              .innerHTML.split(" '")[1]
-              .split("'.")[0];
-            response.message = errorMessage2;
+
+            var serverReply = sentToSpira.errorMessage;
+            var regExp = new RegExp(/<Message>(.*?)<\/Message>/i).exec(serverReply);;
+            response.message = regExp[regExp.length - 1];
           }
           catch (err) {
             try {
-              let htmlError = doc.querySelector('string').innerHTML;
-              response.message = htmlError;
+              var serverReply2 = sentToSpira.errorMessage;

```

Comment:
There is a lot of complex code here and it is not clear what it is doing. Can you add a couple of explainer comments?